### PR TITLE
Code Cleanup and QOL Changes

### PR DESCRIPTION
**Pull request by acd124**
Clean up duplicate code, redundant checks, and messy code
Allow for more user friendly role arguments
TODO: fix callback hell in commands, this requires knowledge of the return types of the current callbacks, recommend that await is used instead

### CODE REVIEW

**Comment by navi1995** [comment]
Hey man,

Just about to go through this and check it all but just wanted to say thank you for contributing! Seems like an amazing amount of QOL changes and improvements made so thank you for taking the time to do it. Will merge in a bit and deploy during some downtime.

File: Discord Bot/bot.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const client = new Discord.Client({
 	messageCacheMaxSize: 50,
 	messageCacheLifetime: 7300, //Maximum poll time = 7200, ensure message not swept.
 	messageSweepInterval: 600,
+	allowedMentions: { parse: ['users'] }, // allowedMentions to prevent unintended role and everyone pings
```
Comment (navi1995) [inline Discord Bot/bot.js:16]: Checked through database and most custom announcement/poll messages contain some sort of role. Don't believe we should limit the bot from being able to mention them.

Comment (acd124) [reply Discord Bot/bot.js:16]: Sure, can allow any mentions when sending the poll message

**Comment by acd124** [comment]
One thing to note is I made a change to the data stored for a movie submittedBy property, the new code stores just the user id, so you would have to update all the old database entries to cut out the <@> in order for it to work properly again

File: Discord Bot/commands/get.js [hunk 2]
```diff
@@ -5,30 +5,25 @@ module.exports = {
...
-
-					return callback();
-				} else if (movies && movies.length > 0) {
-					for (var movie of movies) {
-						var stringConcat = `**[${number}. ${movie.name}](https://www.imdb.com/title/${movie.imdbID})** submitted by ${movie.submittedBy} on ${moment(movie.submitted).format("DD MMM YYYY")}\n
+				if (!movie || !movies.length) { 
+					return message.channel.send("List of unviewed movies is currently empty.");
+				} else {
+					for (let movie of movies) {
+						let stringConcat = `**[${number++}. ${movie.name}](https://www.imdb.com/title/${movie.imdbID})** submitted by <@${movie.submittedBy}> on ${moment(movie.submitted).format("DD MMM YYYY")}\n
```
Comment (navi1995) [inline Discord Bot/commands/get.js:26]: Love this solution but could we do something to make this backwards compatible with current data structure of submitted by? Would be nice not having to update thousands of entries in the database. 

Such as surrounding with <@{}> the submitted by string doesn't contain <@.

Comment (navi1995) [reply Discord Bot/commands/get.js:26]: Actually for simplicity sake I'll revert the changes to submittedBy after merging and re-implement them in a later release after rolling out API change for the dashboard and creating a mingration script for the mongoDB.

**Comment by navi1995** [comment]
Had to make some changes in poll.js and other places that had some errors due to misplaced code but overall a great improvement. This has been rolled out to the live bot if you wish to do any testing on top of what I've done :) Thanks again for your work.

**Comment by acd124** [comment]
`if (movies.length == 0 || !movies.length) {`
line 22 in get.js, This change actually reintroduces redundancy. I had it checking `!movies` to check if movies was missing, or `!movies.length` to check if the length of movies was 0, changing the first one just removes the check for if movies is missing, making it check twice if the length is 0

**Comment by navi1995** [comment]
Ah I see, I'll update this. It was actually checking if (!movie || movies.length) but movie did not exist so it must've been a typo.

**Pull request merged**